contracts/PepemonCardDeck.sol

```diff
@@ -74,48 +80,97 @@ contract PepemonCardDeck is ERC721, ERC1155Holder, Ownable {
+        uint256 maxInitialSupportCards
+    ) public onlyAdmin {
+        require(Arrays.isSortedAscending(battleCardIds));
+        require(maxInitialSupportCards <= MAX_SUPPORT_CARDS);
+
+        initialDeckSupportCardAmount = maxInitialSupportCards;
+        allowedInitialDeckBattleCards = battleCardIds;
+        allowedInitialDeckSupportCards = supportCards;
+    }
+
     // ALLOW TEST MINTING
-    function setMintingCards(uint256 minCardId, uint256 maxCardId) public onlyOwner {
+    function setMintingCards(uint256 minCardId, uint256 maxCardId) public onlyAdmin {
         maxMintTestCardId = maxCardId;
         minMintTestCardId = minCardId;
     }
+
     /**
-     * @dev right now there are 40 different cards that can be minted, but the maximum is configurable with maxMintTestCard. 
+     * @dev right now there are 40 different cards that can be minted, but the maximum is configurable with maxMintTestCard.
      * setting maxMintTestCard to 0 disables this card minting.
      */
     function mintCards() public {
         require(maxMintTestCardId > 0, "Minting test cards is disabled");
-        IPepemonFactory(supportCardAddress).batchMint(minMintTestCardId, maxMintTestCardId, msg.sender);
+        IPepemonFactory(factoryAddress).batchMint(minMintTestCardId, maxMintTestCardId, msg.sender);
+    }
+
+    function mintInitialDeck(uint256 battleCardId) public {
+        require(Arrays.contains(allowedInitialDeckBattleCards, battleCardId), "Invalid battlecard");
+        require(playerToDecks[msg.sender].length == 0, "Not your first deck");
+        // battlecard + support cards
+        uint amount = initialDeckSupportCardAmount + 1;
+        uint256[] memory cards = new uint256[](amount);
+
+        // First step: Mint cards
+
+        uint256 allowedCardsCount = allowedInitialDeckSupportCards.length;
+        uint256 randomNumber = randSeed();
+        cards[0] = battleCardId;
+        // begin from index 1 instead of 0 because battlecard was already added
+        for (uint256 i = 1; i < amount; ++i) {
+            randomNumber = uint256(keccak256(abi.encodePacked(i, randomNumber)));
+            cards[i] = allowedInitialDeckSupportCards[randomNumber % allowedCardsCount];
+        }
+        // mint cards directly for this contract instead of msg.sender, so that we dont have to
+        // transfer it back and forth
+        IPepemonFactory(factoryAddress).batchMintList(cards, address(this));
+
+        // Second step: Add cards into new the deck
+
+        uint256 newDeckId = createDeckInternal();
+
+        // no need to call addBattleCardToDeck since the new deck never had a battlecard before, plus
+        // the battlecard is already owned by this contract
+        decks[newDeckId].battleCardId = battleCardId;
+
+        // begin from index 1 again because battlecard is in index 0
+        for (uint256 i = 1; i < amount; ++i) {
+            addSupportCardToDeckDirectly(newDeckId, cards[i], 1);
```

Comment:
I think this might be problematic. It will create decks that have cards that cannot be removed because the deck doesn't own enough of them.

(removing cards transfers from the deck contract to the deck owner)